contents/docs/tutorials/feature-flags.md

```diff
@@ -0,0 +1,206 @@
+
+```js
+// Header text is blue by default
+// If flag is on, make the text red
+if (posthog.isFeatureEnabled('red-header-text')) {
+    document.getElementById('header-text').style['color'] = 'red'
+}
+```
+
+While this is a trivial example, you can have as much logic as you want being powered by the feature flag. You can also combine flags with `OR` & `AND` operations to create more complex logic.
+
+## Advanced Controls
+
+### Loading Flags
+
+Every time a user loads a page we send a request in the background to an endpoint to get the feature flags that apply to that user. We store those flags as a cookie. 
+
+This means that for most page views the feature flags will be available immediately, *except* for the very first time a user visits your site.
+
+To combat that, there's a callback you can use to wait for the flags to come in:
+
+```js
+posthog.onFeatureFlags(function() {
+    // feature flags are guaranteed to be available at this point
+    if (posthog.isFeatureEnabled('new-beta-feature')) {
+        // do something
+    }
+})
+```
+
+In our JS Integration, the reason we store flags as a cookie is to reduce the load on the server and improve the performance of your app, as it doesn't need to always make an HTTP request, it can simply refer to data stored locally in the browser. 
+
+While this makes your app faster, it means that if your user does something mid-session which causes the flag to turn on for them, this will not be immediately updated. As such, if you expect your app to have scenarios like this _and_ you want flags to update mid-session, you can reload them yourself, by using the `reloadFeatureFlags` function.
+
+```js
+posthog.reloadFeatureFlags()
+```
+
+Calling this function will force PostHog to hit the endpoint for the updated information, and allows you ensure changes are reflected mid-session.
+
+## Feature Flag Uses
+
+Feature flags are a very powerful piece of functionality that can be used in a wide variety of ways. How you use them will depend on your particular painpoints and internal best practices.
+
+For us, here are some suggestions of use cases that could fit nicely with feature flags:
+
+### A/B Testing
+
+![A/B Testing Example](../../images/tutorials/feature-flags/ab-testing.png)
+
+<br />
+
+In simple terms, A/B testing is a method for determining how to provide the best user experience or meet other product goals by testing how different features perform. 
+
+This could be used to answer questions such as "Do users click a button more often if it is blue or red?" or more complex questions like "How much more time do active users who have signed up for a free trial spend on our blog pages if we add a banner image?".
+
+At PostHog, we're still working to provide built-in specific A/B Testing functionality. However, with feature flags, there's already a lot you can do in this regard.
```

Comment:
This can probably come out. Feature flags provide a/b testing- unless you can think of a counterexample to what is possible with the platform?

Reply:
I thought so too - but A/B testing is on our "Future" roadmap for some reason: https://github.com/orgs/PostHog/projects/1

Reply:
I've suggested a change to the title https://github.com/PostHog/posthog/issues/1697 to reflect the reality that you *can* do a/b tests with feature flags but they're not totally intuitive for many GA A/B test users, which is where most people learn about them.